test/integration/webhook/v1alpha1/suite_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	kueuev1alpha1 "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/test/integration/framework"
+	// +kubebuilder:scaffold:imports
+)
+
+var (
+	cfg       *rest.Config
+	k8sClient client.Client
+	fwk       *framework.Framework
+	ctx       context.Context
+)
+
+func TestAPIs(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	ginkgo.RunSpecs(t,
+		"Webhook Suite",
+	)
+}
+
+var _ = ginkgo.BeforeSuite(func() {
+	fwk = &framework.Framework{
+		CRDPath:     filepath.Join("..", "..", "..", "..", "config", "crd", "bases"),
+		WebhookPath: filepath.Join("..", "..", "..", "..", "config", "webhook"),
+		ManagerSetup: func(mgr manager.Manager, ctx context.Context) {
+			err := (&kueuev1alpha1.Workload{}).SetupWebhookWithManager(mgr)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		},
+	}
+	fwk.TweakManagerOptions = func(opts *ctrl.Options) {
+		// Assume testEnv has been started
+		webhookInstallOptions := fwk.GetTestEnv().WebhookInstallOptions
+		opts.Host = webhookInstallOptions.LocalServingHost
+		opts.Port = webhookInstallOptions.LocalServingPort
+		opts.CertDir = webhookInstallOptions.LocalServingCertDir
+
+	}
+	ctx, cfg, k8sClient = fwk.Setup()
+
+	webhookOpts := &fwk.GetTestEnv().WebhookInstallOptions
+	// wait for the webhook server to get ready
+	dialer := &net.Dialer{Timeout: time.Second}
+	addrPort := fmt.Sprintf("%s:%d", webhookOpts.LocalServingHost, webhookOpts.LocalServingPort)
+	gomega.Eventually(func() error {
+		conn, err := tls.DialWithDialer(dialer, "tcp", addrPort, &tls.Config{InsecureSkipVerify: true})
+		if err != nil {
+			return err
+		}
+		conn.Close()
+		return nil
+	}).Should(gomega.Succeed())
```

Comment:
can we move the lines from 74 to 85 to inside fwk.Setup? we an invoke this inside setup `if len(f.WebhookPath) > 0`